rqueue-spring-boot-starter/src/test/java/com/github/sonus21/rqueue/spring/boot/tests/integration/BootDoNotRetryTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019-2024 Sonu Kumar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+package com.github.sonus21.rqueue.spring.boot.tests.integration;
+
+import static com.github.sonus21.rqueue.utils.TimeoutUtils.waitFor;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.github.sonus21.rqueue.exception.TimedOutException;
+import com.github.sonus21.rqueue.spring.boot.application.Application;
+import com.github.sonus21.rqueue.spring.boot.tests.SpringBootIntegrationTest;
+import com.github.sonus21.rqueue.test.dto.Job;
+import com.github.sonus21.rqueue.test.tests.RetryTests;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+
+@ContextConfiguration(classes = Application.class)
+@SpringBootTest
+@Slf4j
+@TestPropertySource(
+    properties = {
+        "rqueue.retry.per.poll=1000",
+        "spring.data.redis.port=8016",
+        "reservation.request.dead.letter.consumer.enabled=true",
+        "reservation.request.active=true",
+        "list.email.queue.enabled=true",
+        "mysql.db.name=BootRetryTest",
+        "use.system.redis=false",
+        "donot.retry=true"
+    })
+@SpringBootIntegrationTest
+class BootDoNotRetryTest extends RetryTests {
+
+  @Test
+  void taskIsNotRetried() throws TimedOutException {
```

Comment:
I'm not sure this test is correct: 
Even though the job throws a DoNotRetryException exception, I can see that the job is being executed several (I think 4) times.

Reply:
let me investigate this, not sure how did it pass. It should have failed